src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -27,6 +28,134 @@ Onyx.connect({
+        const isFirst = index === 0;
+
+        // If any update's previousUpdateID doesn't match the lastUpdateID from the previous update, the deferred updates aren't chained and there's a gap.
+        // For the first update, we need to check that the previousUpdateID of the fetched update is the same as the lastUpdateIDAppliedToClient.
+        // For any other updates, we need to check if the previousUpdateID of the current update is found in the deferred updates.
+        // If an update is chained, we can add it to the applicable updates.
+        const isChained = isFirst ? update.previousUpdateID === lastUpdateIDAppliedToClient : updates[Number(update.previousUpdateID)];
+        if (isChained) {
+            // If a gap exists already, we will not add any more updates to the applicable updates.
+            // Instead, once there are two chained updates again, we can set "firstUpdateAfterGaps" to the first update after the current gap.
+            if (gapExists) {
+                // If "firstUpdateAfterGaps" hasn't been set yet and there was a gap, we need to set it to the first update after all gaps.
+                if (!firstUpdateAfterGaps) {
+                    firstUpdateAfterGaps = Number(update.previousUpdateID);
+                }
+            } else {
+                // If no gap exists yet, we can add the update to the applicable updates
+                applicableUpdates[Number(update.lastUpdateID)] = update;
+            }
+        } else {
+            // When we find a (new) gap, we need to set "gapExists" to true and reset the "firstUpdateAfterGaps" variable,
+            // so that we can continue searching for the next update after all gaps
+            gapExists = true;
+            firstUpdateAfterGaps = undefined;
+
+            // If there is a gap, it means the previous update is the latest missing update.
+            latestMissingUpdateID = Number(update.previousUpdateID);
+        }
+    }
+
+    // When "firstUpdateAfterGaps" is not set yet, we need to set it to the last update in the list,
+    // because we will fetch all missing updates up to the previous one and can then always apply the last update in the deferred updates.
+    if (!firstUpdateAfterGaps) {
+        firstUpdateAfterGaps = Number(updateValues[updateValues.length - 1].lastUpdateID);
+    }
+
+    let updatesAfterGaps: DeferredUpdatesDictionary = {};
+    if (gapExists && firstUpdateAfterGaps) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        updatesAfterGaps = Object.fromEntries(Object.entries(updates).filter(([lastUpdateID]) => Number(lastUpdateID) >= firstUpdateAfterGaps!));
+    }
+
+    return {applicableUpdates, updatesAfterGaps, latestMissingUpdateID};
+}
+
+// This function will check for gaps in the deferred updates and
+// apply the updates in order after the missing updates are fetched and applied
+function validateAndApplyDeferredUpdates(): Promise<Response[] | void> {
```

Comment:
I think a few examples or test cases here will very helpful to understand the code.